src/bp/core/services/dialog/engine.ts

```diff
@@ -72,7 +73,9 @@ export class DialogEngine {
     }
 
     try {
+      await converseApiEvents.emitAsync(`action.start.${event.target}`, event)
       const result = await this.instructionProcessor.process(botId, instruction, event)
+      await converseApiEvents.emitAsync(`action.end.${event.target}`, event)
```

Comment:
I'd put that in the finally of a try/catch so that if it fails it still calls it